overlord/ifacestate/handlers.go

```diff
@@ -227,6 +227,24 @@ func (m *InterfaceManager) doConnect(task *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
+	// connect OS snap if snap name is omitted
+	if slotRef.Snap == "" {
+		slotRef.Snap = "ubuntu-core"
+	}
+
+	// connect the plug to slot with matching interface if slot is omitted
+	if slotRef.Name == "" {
+		plugIface := m.repo.Plug(plugRef.Snap, plugRef.Name)
```

Comment:
Ok, moved to repository.Connect and made it return plugref and slotref. Now, I realize this can be further simplified  a little bit by using your ConnRef from the upcoming disconnect branch (or even only return slotref since plugref doesn't change), but I wanted to keep it simple for this PR and when disconnect lands, I can change it to use ConnRef.